### PR TITLE
M2-340 Remove access for managers

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.const.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { TFunction } from 'i18next';
+import { HeadCell } from 'types/table';
```
Comment (IhorNesterenko99) [inline src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.const.tsx:2]: Pls add empty space line

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.styles.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Box } from '@mui/material';
```
Comment (IhorNesterenko99) [inline src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.styles.ts:1]: Could you pls move these to global styled components to reuse them on other components?

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.const.tsx (hunk 1 above)
Comment (kdanilenka32) [reply src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.const.tsx:3]: pls import t from i18n.ts file

File: src/components/Tables/ManagersTable/ManagersTable.const.tsx [hunk 3]
```diff
@@ -32,10 +32,23 @@ export const getHeadCells = (): HeadCell[] => [
...
-    action: editAccessAction,
-    toolTipTitle: t('editAccess'),
-  },
+export const getActions = (
+  isOwner: boolean,
+  id: string | undefined,
+  { removeAccessAction, editAccessAction }: Actions,
+) => [
+  ...(isOwner || id
+    ? []
```
Comment (kdanilenka32) [inline src/components/Tables/ManagersTable/ManagersTable.const.tsx:41]: why nested array

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  const { t } = useTranslation('app');
+  const { firstName, lastName, email } = user;
+  const applets = user.appletIds.map((id: string) => folders.useApplet(id));
+  const [buttonText, setButtonText] = useState('removeAccess');
+  const [step, setStep] = useState<keyof typeof screenByStep>(1);
+  const [selectedApplets, setSelectedApplets] = useState<string[]>([]);
+  const { control, getValues } = useForm({});
+
+  const getAppletName = useCallback(
+    (id: string) => applets.find((applet) => applet.id === id)?.name || '',
```
Comment (kdanilenka32) [inline src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx:38]: not sure about  useCallback here

File: src/components/Tables/ManagersTable/ManagersTable.const.tsx (hunk 3 above)
Comment (wildkiomi) [reply src/components/Tables/ManagersTable/ManagersTable.const.tsx:41]: not nested. spreaded

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    },
+  }));
+
+  const { execute } = useAsync(revokeAppletUserApi);
+
+  const onSubmit = () => {
+    if (step === 1) {
+      const appletsIds = Object.entries(getValues())
+        .filter((entry) => entry[1])
+        .map(([key]) => key);
```
Comment (kdanilenka32) [inline src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx:71]: use reduce

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx (hunk 4 above)
Comment (kdanilenka32) [reply src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx:41]: no memoization here

File: src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+          {t('userHasAccess')}
+        </StyledBodyLarge>
+        <Table columns={getHeadCells(t)} rows={rows} orderBy="name" uiType={UiType.secondary} />
+      </form>
+    ),
+    2:
+      selectedApplets.length === 1 ? (
+        (() => {
+          const appletName = getAppletName(selectedApplets[0]);
+
```
Comment (kdanilenka32) [inline src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx:115]: pls move steps to components or func. Also you can use array instead object if you have just 1,2,3 keys

File: src/components/Tables/ManagersTable/ManagersTable.const.tsx (hunk 3 above)
Comment (kdanilenka32) [reply src/components/Tables/ManagersTable/ManagersTable.const.tsx:41]: do just like this isOwner || id ? [] : [....]

**Review by IhorNesterenko99 — APPROVED** [review]

File: src/components/Tables/Table/Table.tsx [hunk 7]
```diff
@@ -67,14 +68,14 @@ export const Table = ({
   );
 
   return (
-    <StyledTableContainer height={tableHeight}>
-      <MuiTable stickyHeader>
+    <StyledTableContainer height={tableHeight} uiType={uiType}>
+      <MuiTable stickyHeader={uiType === UiType.primary}>
```
Comment (nmialik) [inline src/components/Tables/Table/Table.tsx:72]: do we need to scroll the head for this uiType?

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**